test/intl402/Intl/getCanonicalLocales/weird-cases.js

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016 Mozilla Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: sec-intl.getcanonicallocales
+description: Tests the getCanonicalLocales function for weird tags.
+info: |
+  8.2.1 Intl.getCanonicalLocales (locales)
+  1. Let ll be ? CanonicalizeLocaleList(locales).
+  2. Return CreateArrayFromList(ll).
+includes: [compareArray.js]
+---*/
+
+var weirdCases =
+  [
+   "x-u-foo",
+   "en-x-u-foo",
+   "en-a-bar-x-u-foo",
+   "en-x-u-foo-a-bar",
+   "en-a-bar-u-baz-x-u-foo",
+  ];
```

Comment:
can we approximate these cases to something that looks close to locales but are not?

Examples: `en-us-`, `-en-us`, `enus`, `en-us-en-us`.

I believe these examples act as negative cases for `IsStructurallyValidLanguageTag`, but please correct me if I'm wrong.